kernel/src/net/socket/ip/stream/mod.rs

```diff
@@ -511,11 +511,14 @@ impl Socket for StreamSocket {
 
     fn shutdown(&self, cmd: SockShutdownCmd) -> Result<()> {
         let state = self.read_updated_state();
-        match state.as_ref() {
-            State::Connected(connected_stream) => connected_stream.shutdown(cmd),
+        let res = match state.as_ref() {
+            State::Connected(connected_stream) => connected_stream.shutdown(cmd, &self.pollee),
             // TODO: shutdown listening stream
             _ => return_errno_with_message!(Errno::EINVAL, "cannot shutdown"),
-        }
+        };
+        drop(state);
+        poll_ifaces();
```

Comment:
`poll_ifaces` should be in a standalone code block, something like:
https://github.com/asterinas/asterinas/blob/43893af043f5c8274c8df4eecf79b6602001f130/kernel/src/net/socket/ip/stream/mod.rs#L238-L242